File: CS/WindowsApplication1/InplaceEditorHelper/DrawEditorHelper.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Columns;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.Drawing;
using DevExpress.Utils.Drawing;
using DevExpress.XtraGrid.Views.Grid;

namespace WindowsApplication1
{
    public static class DrawEditorHelper
    {
        public static void DrawEdit(Graphics g, RepositoryItem edit, Rectangle r, object value)
        {
            BaseEditViewInfo info = edit.CreateViewInfo();
            info.EditValue = value;
            info.Bounds = r;
            info.CalcViewInfo(g);
            ControlGraphicsInfoArgs args = new ControlGraphicsInfoArgs(info, new GraphicsCache(g), r);
            edit.CreatePainter().Draw(args);
            args.Cache.Dispose();
        }

        public static Rectangle GetEditorBounds(Rectangle totalBounds, int rightIdent)
        {
            int leftIndent = 2;
            int topIndent = 2;
            Rectangle bounds = totalBounds;
            bounds.Inflate(-leftIndent, -topIndent);
            bounds.Width -= rightIdent;
            return bounds;
        }
        public static void DrawColumnInplaceEditor(ColumnHeaderCustomDrawEventArgs e, RepositoryItem item, object value, int rightIndent)
        {
            Rectangle targetRect = GetEditorBounds(e.Bounds, rightIndent);
            DrawEdit(e.Graphics, item, targetRect, value);
        }
    }
}

File: VB/WindowsApplication1/InplaceEditorHelper/DrawEditorHelper.vb
Imports System.ComponentModel
Imports System.Drawing
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.Utils.Drawing
Imports DevExpress.XtraGrid.Views.Grid

Namespace WindowsApplication1

    Public Module DrawEditorHelper

        Public Sub DrawEdit(ByVal g As Graphics, ByVal edit As RepositoryItem, ByVal r As Rectangle, ByVal value As Object)
            Dim info As BaseEditViewInfo = edit.CreateViewInfo()
            info.EditValue = value
            info.Bounds = r
            info.CalcViewInfo(g)
            Dim args As ControlGraphicsInfoArgs = New ControlGraphicsInfoArgs(info, New GraphicsCache(g), r)
            edit.CreatePainter().Draw(args)
            args.Cache.Dispose()
        End Sub

        Public Function GetEditorBounds(ByVal totalBounds As Rectangle, ByVal rightIdent As Integer) As Rectangle
            Dim leftIndent As Integer = 2
            Dim topIndent As Integer = 2
            Dim bounds As Rectangle = totalBounds
            bounds.Inflate(-leftIndent, -topIndent)
            bounds.Width -= rightIdent
            Return bounds
        End Function

        Public Sub DrawColumnInplaceEditor(ByVal e As ColumnHeaderCustomDrawEventArgs, ByVal item As RepositoryItem, ByVal value As Object, ByVal rightIndent As Integer)
            Dim targetRect As Rectangle = GetEditorBounds(e.Bounds, rightIndent)
            DrawEdit(e.Graphics, item, targetRect, value)
        End Sub
    End Module
End Namespace
